File: 10_laba/C##/AirTicketOffice/AirTicketOfficeTest/by/bntu/fitr/java/ingoslingsfootsteps/model/logic/CustomerControllerTest.cs
using AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.model.entity;
using AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.model.logic;
using AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.util.exception;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Text;

namespace AirTicketOfficeTest.by.bntu.fitr.java.ingoslingsfootsteps.model.logic
{
    [TestFixture]
    class CustomerControllerTest
    {
        [Test]
        public void addNewCustomerOrGetExisting() {
            CustomerController customerController = new CustomerController();
           Assert.Throws<InputException>(()=>customerController.AddNewCustomerOrGetExisting("", ""));
        }

        [Test]
        public void getExistingCustomerByEmail() {
            Customer isActual = new Customer("Helena", "lenka_bokshic");
            CustomerController customerController = new CustomerController();
            Customer expected = customerController.AddNewCustomerOrGetExisting(isActual.Name, isActual.Email);
            Assert.AreEqual(expected.Email, isActual.Email);
        }

        [Test]
        public void getExistingCustomerByName() {
            Customer isActual = new Customer("Helena", "lenka_bokshic");
            CustomerController customerController = new CustomerController();
            Customer expected = customerController.AddNewCustomerOrGetExisting(isActual.Name, isActual.Email);
            Assert.AreEqual(expected.Name, isActual.Name);
        }

        [Test]
        public void addNewCustomer() {
            Customer isActual = new Customer("Ivan", "Ivanov");
            CustomerController customerController = new CustomerController();
            Customer expected = customerController.AddNewCustomerOrGetExisting(isActual.Name, isActual.Email);
            Assert.AreEqual(expected.Name, isActual.Name);
        }
    }
}

File: 10_laba/test/by/bntu/fitr/java/inGoslingsFootsteps/Lab10/model/logic/CustomerControllerTest.java
package by.bntu.fitr.java.inGoslingsFootsteps.Lab10.model.logic;

import by.bntu.fitr.java.inGoslingsFootsteps.Lab10.model.entity.Customer;
import by.bntu.fitr.java.inGoslingsFootsteps.Lab10.util.exception.InputException;
import org.junit.Test;

import java.io.FileNotFoundException;

import static org.junit.Assert.assertEquals;

public class CustomerControllerTest {

    @Test(expected = InputException.class)
    public void addNewCustomerOrGetExisting() throws FileNotFoundException, InputException {
        CustomerController customerController = new CustomerController();
        customerController.addNewCustomerOrGetExisting("","");
    }

    @Test
    public void getExistingCustomerByEmail() throws FileNotFoundException, InputException {
        Customer isActual = new Customer("Helena", "lenka_bokshic");
        CustomerController customerController = new CustomerController();
        Customer expected = customerController.addNewCustomerOrGetExisting(isActual.getName(), isActual.getEmail());
        assertEquals(expected.getEmail(), isActual.getEmail());
    }

    @Test
    public void getExistingCustomerByName() throws FileNotFoundException, InputException {
        Customer isActual = new Customer("Helena", "lenka_bokshic");
        CustomerController customerController = new CustomerController();
        Customer expected = customerController.addNewCustomerOrGetExisting(isActual.getName(), isActual.getEmail());
        assertEquals(expected.getName(), isActual.getName());
    }

    @Test
    public void addNewCustomer() throws FileNotFoundException, InputException {
        Customer isActual = new Customer("Ivan", "Ivanov");
        CustomerController customerController = new CustomerController();
        Customer expected = customerController.addNewCustomerOrGetExisting(isActual.getName(), isActual.getEmail());
        assertEquals(expected.getName(), isActual.getName());
    }
}